api/app/services/rag/document_retriever.py

```
@@ -474,111 +632,104 @@ def _lc_with_retrieved_id(r: RetrievedDocument) -> Document:
                 unique_docs.append(doc)
-                # Qdrant returns similarity-like scores (higher is better). Keep a safe clamp.
-                similarity = max(0.0, min(1.0, float(score)))
-                unique_scores.append(similarity)
+                retrieval_scores.append(self._clamp_unit_interval(float(score)))
+
+        if self.reranker and unique_docs:
+            try:
+                rerank_top_n = self.rerank_top_n or len(unique_docs)
+                rerank_candidates = [
+                    RetrievedDocument.from_langchain_document(doc, score=score)
+                    for doc, score in zip(unique_docs, retrieval_scores)
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)